apps/gateway/src/middlewares/models/index.ts

```diff
@@ -0,0 +1,61 @@
+import { t, type Static } from "elysia";
+
+import { CohereEmbedV4Adapter } from "./cohere";
+import {
+  Gemini25FlashPreviewAdapter,
+  Gemini25FlashLitePreviewAdapter,
+} from "./gemini";
+import { GptOss120bAdapter, GptOss20bAdapter } from "./gpt";
+
+import type { ModelAdapter } from "./model";
+
+export const supportedModel = t.Object({
+  id: t.String(),
+  name: t.String(),
+  created: t.Number(),
+  owned_by: t.String(),
+  modality: t.Union([t.Literal("chat"), t.Literal("embedding")]),
+  pricing: t.Object({
+    monthly_free_tokens: t.Number(),
+  }),
+});
+
+export type SupportedModel = Static<typeof supportedModel>;
+
+const MODEL_ADAPTER_MAP = {
```

Comment:
Can you derive this from the Adapters itself, instead of maintaining a second map, same as you do with the Providers?